### DISCUSSION TITLE
WunderLINQ long press right starts follow location

### DISCUSSION THREAD

**Discussion by ToToGSARider** [devemux86/cruiser#632] [Q&A] [answered]
Hello,

I start the Cruiser app from the WunderLINQ app. Currently I have to tap the Follow Mode manually. Is it possible to teach the Cruiser app to start follow mode by pressing right, short or long, when no navigation is running?

Hallo,

Ich starte ja aus der WunderLINQ App die Cruiser App. Aktuell muss ich den Follow Mode von Hand antippen. Ist es möglich, der Cruiser App beizubringen, wenn keine Navigation läuft, durch drücken nach rechts, kurz oder lang, den Follow Mode zu starten?

**Comment by devemux86** [comment]
@ToToGSARider 

Does WunderLINQ support long press? We need to ask @blackboxembedded.
Also the left / right is used in WunderLINQ (move) option.

Unterstützt WunderLINQ langes Drücken? Wir müssen @blackboxembedded fragen.
Auch die links / rechts Option wird in WunderLINQ (Verschieben) verwendet.

---

@blackboxembedded

Is there an updated document for the available key codes of WunderLINQ?

And do they work with long press? How?

**Reply by blackboxembedded** [reply]
Sorry, just saw this. This is the latest document: https://cdn.shopify.com/s/files/1/0052/6195/1025/files/WunderLINQ_3rd_Party_Integration.pdf?v=1594610574

**Reply by devemux86** [reply]
@blackboxembedded thanks!

**Reply by blackboxembedded** [reply]
@devemux86 Thanks for the WunderLINQ enhancements!

**Comment by ToToGSARider** [comment]
So the thing is that when I say: navigate to home in the WunderLINQ app, the destination is transferred and confirmed by a long press. 
Pushing the map is a separate function in Cruiser. When I was driving and long pressed to the right in follow mode, nothing happened. 
If I start the Cruiser app from the WunderLINQ app without Follow mode, nothing happens when I long press to the right. Then the command should be available there?


Also es ist ja so, dass wenn ich in der WunderLINQ App sage: zu Heimat navigieren, wird das Ziel ja übergeben und durch langes drücken bestätigt. 
Das schieben der Karte ist ja in Cruiser eine getrennte Funktion. Als ich eben gefahren bin und im Folgemodus lange nach rechts gedrückt habe, ist nichts passiert. 
Wenn ich aus der WunderLINQ App die Cruiser App starte, ohne Follow Modus, passiert auch nichts wenn ich lange nach rechts drücke. Dann sollte der Befehl doch dort frei sein?

**Reply by devemux86** [reply]
@ToToGSARider 

When select "WunderLINQ (Zoom)", then the right / left are available.
When select "WunderLINQ (Move)", then the right / left are in use.

Anyway I need documentation from @blackboxembedded to understand what is possible.

Wenn Sie "WunderLINQ (Zoom)" auswählen, sind die Tasten rechts / links verfügbar.
Wenn man "WunderLINQ (Verschieben)" auswählt, dann sind rechts / links in Gebrauch.

Auf jeden Fall brauche ich Dokumentation von @blackboxembedded zu verstehen, was möglich ist.

**Comment by ToToGSARider** [comment]
Hier ist der Link zur Internetseite, wo es eine Dokumentation gibt:

https://blackboxembedded.com/pages/developers

https://cdn.shopify.com/s/files/1/0052/6195/1025/files/WunderLINQ_3rd_Party_Integration.pdf?v=1594610574

**Comment by devemux86** [comment]
@ToToGSARider 

Thanks, so the available keys are:
- Left, Right, Up, Down
- Escape (long press left)
- Enter (long press right)

Since the Escape key is used to go back to the WunderLINQ app, we can use the Enter key (long press right)?
To start navigation or follow mode?

---

Danke, die verfügbaren Tasten sind also:
- Links, Rechts, Hoch, Runter
- Escape (links lange drücken)
- Enter (rechts lange drücken)

Da die Escape-Taste verwendet wird, um zur WunderLINQ-App zurückzukehren, können wir die Enter-Taste (rechts lange drücken) verwenden?
Um die Navigation oder den Folgemodus zu starten?

**Reply by ToToGSARider** [reply]
We already had a lot of ideas in the thread Skip waypoints. For example, you could also say, short left press: route overview, long right press: start follow mode, short right press: open menu and then confirm the option by pressing right left through the menu and long right.

Wir hatten ja schon einmal im Thread Wegpunkte überspringen auch viele Ideen. Man könnte ja zum Beispiel auch sagen, kurzer Links druck: Routenübersicht, langer rechts druck Follow Modus starten, kurzer rechts druck Menü öffnen und dann dort die Möglichkeit durch rechts links durch das Menü und lange rechts drücken bestätigen.

**Reply by devemux86** [reply]
@ToToGSARider 

Remember that the short left / right keys are only available when selecting "WunderLINQ (Zoom)" to only zoom the map.
Long press right is always available.

Denken Sie daran, dass die kurzen Links-/Rechtstasten nur verfügbar sind, wenn Sie "WunderLINQ (Zoom)" auswählen, um nur die Karte zu zoomen.
Langes Drücken nach rechts ist immer möglich.

**Reply by ToToGSARider** [reply]
I only ever use zoom mode, as we always drive with route recalculation "off". 
The zoom is perfect for detours to see where the route goes. Automatic zoom is also always deactivated.

Ich nutze immer nur den Zoom Modus, da wir immer Routenneuberechnung "aus" fahren. 
Bei Umleitungen ist der Zoom perfekt, um zu sehen wo die Route läuft. Auch der automatische Zoom ist immer deaktiviert.

**Comment by devemux86** [comment]
@ToToGSARider 

Next Cruiser update will allow to use the long right press of WunderLINQ to enable/disable the location follow mode.

Das nächste Cruiser-Update ermöglicht die Verwendung eines langen Rechtsdrucks auf WunderLINQ zum Aktivieren/Deaktivieren des Standortverfolgungsmodus.

**Reply by ToToGSARider** [reply]
The response time is great as always. Reported yesterday and already this feedback today. I'm already looking forward to the next opportunities for the WunderLINQ. It would be nice if they would also appear here in the thread. 

Die Reaktionszeit ist wie immer großartig. Gestern gemeldet und heute schon dieses Feedback. Ich freue mich schon auf die nächsten Möglichkeiten für das WunderLINQ. Wäre schön, wenn die dann auch hier mit im Thread erscheinen würden.

**Reply by devemux86** [reply]
@ToToGSARider 

It is available for testing in the latest:

Es ist spätestens jetzt zum Testen verfügbar:

- https://github.com/devemux86/cruiser/discussions/610

**Reply by ToToGSARider** [reply]
Muß ich mir jetzt Cruiser 5.0 installieren? Kann ich meine Einstellungen aus Cruiser 4 übernehmen in 5? Kann ich Cruiser 5.0 einfach darüber installieren? 

Do I have to install Cruiser 5.0 now? Can I transfer my settings from Cruiser 4 to 5? Can I simply install Cruiser 5.0 over it?

**Reply by devemux86** [reply]
@ToToGSARider 

Cruiser 5 can be installed on 4 and all user data and settings are automatically used.
Although it would be nice to test WunderLINQ, it is still a beta version.
If you are not comfortable with beta testing, better test it on another phone or wait for the release on Google Play.

Cruiser 5 kann auf 4 installiert werden und alle Benutzerdaten und Einstellungen werden automatisch übernommen.
Obwohl es schön wäre, WunderLINQ zu testen, ist es immer noch eine Beta-Version.
Wenn Sie sich mit Beta-Tests nicht wohlfühlen, testen Sie es besser auf einem anderen Telefon oder warten Sie auf die Veröffentlichung auf Google Play.

**Reply by ToToGSARider** [reply]
Hello,

like Rolf GS, I'm waiting for the official version. Maybe you can add more functions by then 🥇 

Hallo,

ich warte dann genau wie Rolf GS auf die offizielle Variante. Vielleicht könnt ihr bis dahin ja noch mehr Funktionen einbauen 🥇

**Comment by devemux86** [comment, marked as answer]
- WunderLINQ long press right starts follow location is available in https://github.com/devemux86/cruiser/discussions/641

**Reply by ToToGSARider** [reply]
Und wenn dann auch irgendwann 3D geht, dann kann ich es ja auch verwenden. Ich hoffe auf 2025.

And if 3D works at some point, then I can use it too. I'm hoping for 2025.

**Comment by ToToGSARider** [comment]
Just tested in the garage to start the Follow Mode directly with the WunderLINQ, it works. Thank you very much for the new version

So eben in der Garage mal getestet direkt den Follow Mode zu starten mit dem WunderLINQ, läuft. Vielen lieben Dank für die neue Version
